### PR TITLE
Feat factories and seeders implementation

### PR DESCRIPTION
**Pull request by purplena**

### CODE REVIEW

File: database/seeders/DatabaseSeeder.php [hunk 1]
```diff
@@ -2,19 +2,51 @@
...
+    $companies = Company::all();
+    foreach ($songs as $song) {
+      $randomCompany = $companies->random();
+      $randomCompany->songs()->attach($song->id);
+    }
+  }
+
+  protected function seedBlacklistedSongs()
+  {
+    $blacklistedSongs = collect(Song::all())->slice(4, 2);
```
Comment (stephdotnet) [inline database/seeders/DatabaseSeeder.php:44]: remove collect()

File: database/seeders/RequestedSongSeeder.php [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+class RequestedSongSeeder extends Seeder
+{
+  public function run(): void
+  {
+    for ($i = 0; $i < 15; $i++) {
+      RequestedSong::create([
+        'song_id' => Song::all()->take(5)->random()->id,
+        'user_id' => User::where('role', 2)->get()->random()->id,
+        'company_id' => Company::all()->random()->id
```
Comment (stephdotnet) [inline database/seeders/RequestedSongSeeder.php:19]: - Takes the queries out.
- You can take in random order : https://laravel.com/docs/10.x/queries#random-ordering
- Use limit() to reduce the volume of taken songs
- Take only songs that are not blacklisted

File: database/seeders/DatabaseSeeder.php (hunk 1 above)
Comment (stephdotnet) [reply database/seeders/DatabaseSeeder.php:44]: Also, you should pick blacklisted song in random order

File: database/seeders/SongSeeder.php [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+use App\Models\Song;
+use Illuminate\Database\Seeder;
+
+class SongSeeder extends Seeder
+{
+  /**
+   * Run the database seeds.
+   */
+  public function run(): void
+  {
```
Comment (stephdotnet) [inline database/seeders/SongSeeder.php:14]: Create a factory for songs. 
You can have an array in the factory for the spotify cover image url

File: database/seeders/UpvoteSeeder.php [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+use Illuminate\Database\Seeder;
+
+class UpvoteSeeder extends Seeder
+{
+  public function run(): void
+  {
+    for ($i = 0; $i < 20; $i++) {
+      Upvote::create([
+        'requested_song_id' => RequestedSong::all()->random()->id,
+        'user_id' => User::where('role', 2)->get()->random()->id,
```
Comment (stephdotnet) [inline database/seeders/UpvoteSeeder.php:17]: Queries out

File: database/seeders/CompanySeeder.php [hunk 5]
```diff
@@ -26,10 +27,11 @@ public function run(): void
 
     Company::factory()
       ->has(User::factory()->owner())
-      ->count(10)
+      // ->hasAttached(Song::all()->take(5))
```
Comment (stephdotnet) [inline database/seeders/CompanySeeder.php:30]: Comment to remove

**Review by stephdotnet — CHANGES_REQUESTED** [review]

File: database/seeders/CompanySeeder.php [hunk 6]
```diff
@@ -26,10 +27,11 @@ public function run(): void
...
+      ->count(2)
       ->create();
-
-    // Company::factory()->count(2)->create();
-    // User::factory()->owner()->create();
   }
+
+  // Company::factory()->count(2)->create();
+  // User::factory()->owner()->create();
```
[the remark this answers is not in the document]
Comment (purplena) [reply database/seeders/CompanySeeder.php:36]: done!

File: database/seeders/CompanySeeder.php (hunk 5 above)
Comment (purplena) [reply database/seeders/CompanySeeder.php:30]: removed

File: database/seeders/DatabaseSeeder.php (hunk 1 above)
Comment (purplena) [reply database/seeders/DatabaseSeeder.php:44]: Integrated a system when we take 5 songs using `limit `to optimize a query  ` $blacklistedSongs = Song::limit(5)->get()`; Then I distribute those songs randomly among companies

File: database/seeders/RequestedSongSeeder.php (hunk 2 above)
Comment (purplena) [reply database/seeders/RequestedSongSeeder.php:19]: Took out all the queries. Make a `DB query` to get all the songs that have not been blacklisted. I continue to `random()` as it is one of the methods of Laravel Collections

File: database/seeders/SongSeeder.php (hunk 3 above)
Comment (purplena) [reply database/seeders/SongSeeder.php:14]: Created a separate factoy for the songs

File: database/seeders/UpvoteSeeder.php (hunk 4 above)
Comment (purplena) [reply database/seeders/UpvoteSeeder.php:17]: Removed queried from the loop

**Review by purplena — COMMENTED** [review]
All comments are closed

**Review by stephdotnet — CHANGES_REQUESTED** [review]
There are some small adjustments to do :) 

Also i noticed you have (i think because of the merge conflict resolution) a duplicate migration on upvotes. The issue is that it breaks the migrations (since upvotes table is already created)

image

File: database/seeders/CompanySongSeeder.php [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+class CompanySongSeeder extends Seeder
+{
+  public function run(): void
+  {
+    $songs = DB::table('songs')
+      ->select('songs.*')
+      ->leftJoin('company_song_blacklisted', 'company_song_blacklisted.song_id', '=', 'songs.id')
+      ->whereNull('company_song_blacklisted.song_id')
+      ->get();
```
[the remark this answers is not in the document]
Comment (stephdotnet) [reply database/seeders/CompanySongSeeder.php:17]: DB queries : https://laravel.com/docs/10.x/queries
Eloquent Querying : https://laravel.com/docs/10.x/eloquent#retrieving-models

**Pull request merged**